### PR TITLE
[OPPRO-293] Enable DateExpressionsSuite test for velox backend

### PR DESCRIPTION
**Pull request by PHILO-HE**

### CODE REVIEW

File: gluten-ut/src/test/scala/org/apache/spark/sql/GlutenTestsTrait.scala [hunk 1]
```diff
@@ -172,7 +171,15 @@ trait GlutenTestsTrait extends SparkFunSuite with ExpressionEvalHelper with Glut
...
     val resultDF = df.select(Column(expression))
-    val result = resultDF.collect()
+    val result = if (justEvalExpr) {
+      try {
+        expression.eval(inputRow)
+      } catch {
+        case e: Exception => fail(s"Exception evaluating $expression", e)
+      }
+    } else {
+      resultDF.collect()
```
Comment (jinchengchenghh) [inline gluten-ut/src/test/scala/org/apache/spark/sql/GlutenTestsTrait.scala:181]: Does `resultDF.collect()` raise error? Can we fix that?

**Review by zzcclp — APPROVED** [review]
LGTM

File: gluten-ut/src/test/scala/org/apache/spark/sql/GlutenTestsTrait.scala (hunk 1 above)
Comment (PHILO-HE) [reply gluten-ut/src/test/scala/org/apache/spark/sql/GlutenTestsTrait.scala:181]: Thanks for your comment! I think it is just a testing issue, not related to the main code functionality. Let's respect vanilla spark's testing approach.

File: gluten-ut/src/test/scala/org/apache/spark/sql/catalyst/expressions/GlutenDateExpressionsSuite.scala [hunk 2]
```diff
@@ -17,7 +17,22 @@
...
+                                         inputRow: InternalRow = EmptyRow): Unit = {
+    val resolver = ResolveTimeZone
+    val expr = resolver.resolveTimeZones(expression)
+    assert(expr.resolved)
+
+    val catalystValue = CatalystTypeConverters.convertToCatalyst(expected)
+    // Consistent with the evaluation approach in vanilla spark UT to avoid overflow issue
+    // in resultDF.collect() for some corner cases.
+    glutenCheckExpression(expr, catalystValue, inputRow, true)
+  }
```
Comment (lviiii) [inline gluten-ut/src/test/scala/org/apache/spark/sql/catalyst/expressions/GlutenDateExpressionsSuite.scala:37]: Does every UT have to override this method  to avoid exception?

Comment (PHILO-HE) [reply gluten-ut/src/test/scala/org/apache/spark/sql/catalyst/expressions/GlutenDateExpressionsSuite.scala:37]: The current dataframe check logic may have some drawbacks to hit some exceptions. We can just do such handling for failure UTs whose root cause is not on our code side.
